### PR TITLE
APPS-1798 update dx run --instance-type-help with --instance-type-by-executable

### PR DESCRIPTION
**Pull request by YuxinShi0423**
This PR is to update dx run --instance-type-help to include instructions using flag `dx run --instance-type-by-executable` 

This PR should be merged after the instance-type-by-executable added to dx run and dx-jobutil-new-job as in PR https://github.com/dnanexus/dx-toolkit/pull/1006

### CODE REVIEW

File: src/python/dxpy/cli/parsers.py [hunk 1]
```diff
@@ -226,23 +226,85 @@ class PrintInstanceTypeHelp(argparse.Action):
...
+                       'for the main entry point and mem2_ssd1_v2_x2 for all other entry points.' +
+                       'Also force the collect entry point of all executables other than applet-xyz1 to use mem2_ssd1_v2_x8.' +
+                       'Other entry points of executable other than applet-xyz1 may be overridden by ' +
+                       'lower-priority mechanisms'))
+            print()
+            print('    dx run workflow-xxxx --instance-type-by-executable \\\n' +
+                  '           \'{"applet-xyz1":  {"main":    "mem2_ssd1_v2_x4", "*": "mem2_ssd1_v2_x2"},\n' +
+                  '             "*":            {"collect": "mem2_ssd1_v2_x8"}}\'')
+            print()
+            print(fill('6. Force every job executing applet-xxxx to use mem2_ssd1_v2_x2 for all entry points' +
```
Comment (commandlinegirl) [inline src/python/dxpy/cli/parsers.py:296]: Please add a space at the end of the line to avoid `pointsin`.

File: src/python/dxpy/cli/parsers.py [hunk 2]
```diff
@@ -226,23 +226,85 @@ class PrintInstanceTypeHelp(argparse.Action):
...
+                       'in the entire execution tree. ' +
+                       'Also force stage 0 executable to run on mem2_ssd1_v2_x4, unless stage 0 invokes' +
```
Comment (commandlinegirl) [inline src/python/dxpy/cli/parsers.py:298]: See my comment to line 296.

**Review by commandlinegirl — APPROVED** [review]
LGTM modulo minor points.

File: src/python/dxpy/cli/parsers.py (hunk 1 above)
Comment (YuxinShi0423) [reply src/python/dxpy/cli/parsers.py]: Fixed. Thanks for catching it!

File: src/python/dxpy/cli/parsers.py (hunk 2 above)
Comment (YuxinShi0423) [reply src/python/dxpy/cli/parsers.py]: Fixed :D

**Pull request merged**